src/sparselm/model/_stepwise.py

```diff
@@ -0,0 +1,164 @@
+"""Composite estimator for piece-wise fitting."""
+
+__author__ = "Fengyu Xie"
+
+from itertools import chain
+
+import numpy as np
+from numpy.typing import ArrayLike
+from sklearn.base import RegressorMixin
+from sklearn.linear_model._base import LinearModel
+from sklearn.utils.metaestimators import _BaseComposition
+
+# BaseComposition makes sure that StepwiseEstimator can be correctly cloned.
+
+
+class StepwiseEstimator(_BaseComposition, RegressorMixin, LinearModel):
+    """A composite estimator used to do stepwise fitting.
+
+    The first estimator in the composite will be used to fit
+    certain features (a piece of the feature matrix) to the
+    target vector, and the residuals are fitted to the rest
+    of features by using the next estimators in the composite.
+    """
+
+    def __init__(self, steps, estimator_feature_indices):
+        """Initialize estimator.
+
+        Args:
+            steps(list[(str, CVXEstimator)]):
+                A list of step names and the CVXEstimators to use
+                for each step. CompositeEstimator cannot be used as
```

Comment:
change Composite to Stepwise in docstrings

Reply:
> > GridsearchCV with stepwise estimator is getting different results from expected. One problem I can see is the handling of the intercept when doing initial fit with intercept. This fit should not include the first column of the feature matrix, corresponding to the constant. Instead the determined intercept should be the coefficient corresponding to the first feature. Perhaps we should take out the zeroth index in the "scope" variable when fitting with intercept. And there would need to be an additional line to add the intercept as the zeroth coef.
> 
> I'm confused what the issue is here @kamronald . Are you passing a feature matrix with a column of all 1s and setting `fit_intercept=True`?
> 
> The estimators have no and should not have knowledge on the feature matrix given so if the feature matrix passed includes a feature of all 1s then the user is responsible to properly set `fit_intercept` accordingly.

Yes I am passing the feature matrix with column of all 1s, because in the current implementation it is required for all columns to be passed.

Reply:
Done.